src/platform/realtek/zephyr/ota/matter_ota.c

```diff
@@ -0,0 +1,480 @@
+            {
+                matter_ota_db->state = MATTER_OTA_STATE_IMAGE_HEADER;
+            }
+            else
+            {
+                matter_ota_db->state = MATTER_OTA_STATE_MP_INFO;
+            }
+        }
+
+        LOG_INF("rtk_matter_ota_process_payload: image offset 0x%x", matter_ota_db->cur_download_size);
+    }
+
+    return true;
+
+fail_check_crc:
+fail_write_data:
+    LOG_ERR("rtk_matter_ota_process_payload: failed %d", -ret);
+    return false;
+}
+
+uint8_t rtk_matter_ota_prepare(void)
+{
+    uint8_t ret = 0;
+
+    matter_ota_db = malloc(sizeof(T_MATTER_OTA_DB));
+    if (matter_ota_db == NULL)
+    {
+        ret = 1;
+        goto fail_alloc_db;
+    }
+
+    matter_ota_db->p_buf = malloc(MATTER_OTA_BUF_SIZE);
+    if (matter_ota_db->p_buf == NULL)
+    {
+        ret = 2;
+        goto fail_alloc_buf;
+    }
+
+    matter_ota_db->state              = MATTER_OTA_STATE_PACK_HEADER;
+    matter_ota_db->buf_offset         = 0;
+    matter_ota_db->p_sub_image_header = NULL;
+    matter_ota_db->sub_image_num      = 0;
+    matter_ota_db->cur_image_index    = 0;
+    matter_ota_db->is_data_valid      = false;
+    matter_ota_db->active_bank        = get_active_bank_num();
+
+    LOG_INF("rtk_matter_ota_prepare: active bank %d", matter_ota_db->active_bank);
+
+    return 0;
+
+fail_alloc_buf:
+    free(matter_ota_db);
+    matter_ota_db = NULL;
+fail_alloc_db:
+    LOG_ERR("rtk_matter_ota_prepare: failed %d", -ret);
+    return ret;
+}
+
+void rtk_matter_ota_clear()
+{
+    if (matter_ota_db)
+    {
+        if (matter_ota_db->p_sub_image_header)
+        {
+            free(matter_ota_db->p_sub_image_header);
+        }
+
+        if (matter_ota_db->p_buf)
+        {
+            free(matter_ota_db->p_buf);
+        }
+
+        free(matter_ota_db);
+        matter_ota_db = NULL;
+    }
+}
+
+uint8_t rtk_matter_ota_finalize(void)
+{
+    uint8_t i;
+
+    for (i = 0; i < matter_ota_db->sub_image_num; i++)
+    {
+        if ((matter_ota_db->active_bank == OTA_BANK0 && i >= matter_ota_db->sub_image_num / 2) ||
+            (matter_ota_db->active_bank == OTA_BANK1 && i < matter_ota_db->sub_image_num / 2))
+        {
+            dfu_set_ready((T_IMG_HEADER_FORMAT *) matter_ota_db->p_sub_image_header[i].download_addr);
+        }
+    }
+
+    rtk_matter_ota_clear();
+
+    LOG_INF("rtk_matter_ota_finalize");
+
+    return 0;
+}
+
+void rtk_matter_ota_apply(void)
+{
+    LOG_INF("rtk_matter_ota_apply");
+
+    dfu_fw_reboot(RESET_ALL, DFU_ACTIVE_RESET);
+}
+
+uint8_t rtk_matter_ota_abort(void)
+{
+    LOG_INF("rtk_matter_ota_abort");
+
+    rtk_matter_ota_clear();
+
+    return 0;
+}
+
+uint8_t rtk_matter_ota_process_block(uint8_t * p_data, uint16_t len)
+{
+    uint8_t ret = 0;
+
+    if (matter_ota_db->state == MATTER_OTA_STATE_MP_INFO)
+    {
+        // just drop mp info to save time
+        return 0;
+    }
+
+    if (!rtk_matter_ota_append(p_data, len))
+    {
+        ret = 1;
+        goto fail_append_data;
```

Comment:
why do we have the gotos and not return values directly?

Use named constants for the return values.

Reply:
Modified